### ISSUE TITLE
Type annotations required for as_ref

### ISSUE THREAD

**Issue by tathanhdinh** [sharkdp/bat#148]
I've found error when compiling bat with `rustc 1.28.0-nightly`:

```
error[E0283]: type annotations required: cannot resolve `std::string::String: std::convert::AsRef<_>`
  --> src/features.rs:50:43
   |
50 |             print!("{}", Green.paint(word.as_ref()));
   |                                           ^^^^^^

error: aborting due to previous error
```

that seems because of changes in type inference on nightly and beta ([others](https://users.rust-lang.org/t/old-crate-of-mine-fails-to-compile-on-beta-nightly/6960) found [similar](https://github.com/rust-lang/rust/issues/42226) problems).

IMHO, that is because of there are two possibilities for type of `word.as_ref()`: `&str` and `&[u8]` ; this can be "fixed" using slice notation, like that:

```
print!("{}", Green.paint(&word[..]));
```

which will compile on both stable and nightly.

**Comment by tathanhdinh** [comment]
Many thanks!
I can make a PR if you don't mind.

**Comment by sharkdp** [comment]
Closed via #153
